src/components/Comments.jsx

```diff
@@ -5,24 +5,28 @@ import Box from "@mui/material/Box";
 import TextField from "@mui/material/TextField";
 import Button from "@mui/material/Button";
 
-import { getComments } from "../utils/api";
+import { getCommentsbyId } from "../utils/api";
 import CommentCard from "./CommentCard";
 
 export default function Comments(props) {
 
     const [comments, setComments] = useState([])
+    const [isLoading, setIsLoading] = useState(true);
 
     const {article_id} = props
 
     useEffect(()=>{
-        getComments(article_id).then((commentsData)=>{
-            setComments(commentsData)
+        setIsLoading(true);
+        getCommentsbyId(article_id)
+        .then((commentsData)=>{
+            setIsLoading(false);
+            setComments(commentsData);
         })
     }, [])
```

Comment:
In your console are you getting a warning about article_id missing from dependancy array? I think react would like it if article_id was in there.